src/tile.cpp

```diff
@@ -825,36 +828,36 @@ Tile* Tile::queryDestination(int32_t&, const Thing&, Item** destItem, uint32_t&
 	}
 
 	if (!destTile) {
-		destTile = this;
+		destTile = getTile();
 	} else {
 		flags |= FLAG_NOLIMIT; // Will ignore that there is blocking items/creatures
 	}
 
 	if (destTile) {
-		Thing* destThing = destTile->getTopDownItem();
+		auto destThing = destTile->getTopDownItem();
 		if (destThing) {
-			*destItem = destThing->getItem();
+			destItem = destThing->getItem();
 		}
 	}
 	return destTile;
 }
 
-void Tile::addThing(Thing* thing) { addThing(0, thing); }
+void Tile::addThing(std::shared_ptr<Thing> thing) { addThing(0, thing); }
 
-void Tile::addThing(int32_t, Thing* thing)
+void Tile::addThing(int32_t, std::shared_ptr<Thing> thing)
 {
-	Creature* creature = thing->getCreature();
+	auto creature = thing->getCreature();
 	if (creature) {
 		g_game.map.clearSpectatorCache();
 		if (creature->getPlayer()) {
 			g_game.map.clearPlayersSpectatorCache();
 		}
 
-		creature->setParent(this);
+		creature->setParent(getTile());
 		CreatureVector* creatures = makeCreatures();
 		creatures->insert(creatures->begin(), creature);
 	} else {
-		Item* item = thing->getItem();
+		auto item = thing->getItem();
```

Comment:
Could this be `const auto`
```suggestion
		const auto item = thing->getItem();
```